lib/galaxy/workflow/steps.py

```diff
@@ -35,7 +35,7 @@ def order_workflow_steps(steps):
     if position_data_available:
         steps.sort(key=lambda _: math.sqrt(_.position['left'] ** 2 + _.position['top'] ** 2))
     try:
-        edges = edgelist_for_workflow_steps(steps)
+        edges = sorted(edgelist_for_workflow_steps(steps))
```

Comment:
@jmchilton does this make sense to you ? I'm a little concerned that by updating the test workflows I have bent the test to pass for the sake of passing ...

Reply:
Hmm... such a good question. The fact that sorted is needed means the previous test was somehow depending on the particular hashing or random lib behavior of Python 2.7 I assume - which is less than ideal. I'd like to establish a "correct order" for the steps in a workflow so ``order_index`` is preserved for structurally similar workflows across saves - the test still passes so it is preserved - its just different than it was previously. ``topsort`` using ``random`` so I guess I shouldn't be surprised I'm failing to establish a correct order - the test probably stinks.

Reply:
I don't think `topsort` itself is the issue. I tried this in isolation using the `pairlist` I intercepted form python 2, it gives consistent results between py2 and py3. I think something is different in `edgelist_for_workflow_steps`, perhaps the order in which `step.input_connections` are listed ?